cmake/Cuda.cmake

```diff
@@ -12,20 +12,23 @@ check_cxx_compiler_flag("-std=c++11"   SUPPORT_CXX11)
 function(mshadow_detect_installed_gpus out_variable)
 set(CUDA_gpu_detect_output "")
   if(NOT CUDA_gpu_detect_output)
+    message(STATUS "Running GPU architecture autodetetion")
     set(__cufile ${PROJECT_BINARY_DIR}/detect_cuda_archs.cu)
 
     file(WRITE ${__cufile} ""
       "#include <cstdio>\n"
+      "#include <iostream>\n"
+      "using namespace std;\n"
       "int main()\n"
       "{\n"
       "  int count = 0;\n"
-      "  if (cudaSuccess != cudaGetDeviceCount(&count)) return -1;\n"
-      "  if (count == 0) return -1;\n"
+      "  if (cudaSuccess != cudaGetDeviceCount(&count)) { return -1;}\n"
+      "  if (count == 0) { cerr << \"No cuda devices detected\" << endl; return -1;}\n"
       "  for (int device = 0; device < count; ++device)\n"
       "  {\n"
       "    cudaDeviceProp prop;\n"
       "    if (cudaSuccess == cudaGetDeviceProperties(&prop, device))\n"
-      "      std::printf(\"%d.%d \", prop.major, prop.minor);\n"
+      "      printf(\"%d.%d \", prop.major, prop.minor);\n"
```

Comment:
why? now it’s possibly in conflict with C version. even if not, why change it?

Reply:
because I imported std. Reverted anyway. Any more comments?

Reply:
@cjolivier01 also I don't get your point, since std::printf is the same as ::printf...   or not?